### DISCUSSION TITLE
Securing workflow_dispatch manual workflow?

### DISCUSSION THREAD

**Discussion by noinarisak** [community/community#26008] [Actions]
<p>Hi</p>
<p>Is there a way to secure workflow_dispatch workflow? I would like only Maintainers to be able to execute manual workflow (eg Deploy to EBS).</p>

**Comment by kingthorin** [comment, marked as answer]
<p>You can check who is triggering the workflow:<br>
<code>    if: github.actor == 'noinarisak' || github.actor == 'kingthorin'</code></p>
<p>Example here: <a href="https://github.com/OWASP/www-project-web-security-testing-guide/blob/53d24199a86ef59888ad8b91d8a173468d862753/.github/workflows/pr_comment.yml#L10" rel="nofollow noopener">https://github.com/OWASP/www-project-web-security-testing-guide/blob/53d24199a86ef59888ad8b91d8a173468d862753/.github/workflows/pr_comment.yml#L10</a><br>
It’s slightly different, it runs for anyone other than the actors we check, but same basic idea. (<code>!=</code> vs <code>==</code>.)</p>
<p>Sadly there isn’t (currently) a way to check if your team or a specific role contains the <code>github.actor</code>. I’ve submitted an enhancement request that they add something like <code>@organization/some-team.contains(github.actor)</code>. Just as I’m writing this reply it occurs to me that something like <code>github.actor.role == 'Maintainer'</code> might also be handy.</p>
<p>Edit: There’s also some more details from GitHub staff here: <a href="https://github.community/t/who-can-manually-trigger-a-workflow-using-workflow-dispatch/128592/4" class="inline-onebox">Who can manually trigger a workflow using workflow_dispatch</a> :</p>
<blockquote>
<p>To trigger a workflow in a repository, the user should be a collaborator with <strong>Write</strong> permission in the repository. Normally the external users can’t trigger workflows in the repository.<br>
If the external users use a pull request to trigger workflow from the forked repository, the <strong>GITHUB_TOKEN</strong> only has <strong>read</strong> permissions for some scope.</p>
</blockquote>

**Comment by noinarisak** [comment]
<p>Thanks <a class="mention" href="/kingthorin">@kingthorin</a>!</p>
<p><code>if:  github.actor = ...</code> example will have to work for now. Your enhancement request would be ideal (eg <code>@organization/some-team.contains(github.actor)</code>), especially because I have to address onboarding and offboarding members and its pain to keep updating the yaml every time. 1. The list of members is just an eyesore and leaves a lot of room for errors. :sweat_smile:</p>
<p>Is the GH Actions enhancement request something I can add my additional vote too?</p>

**Comment by kingthorin** [comment]
<p>Sadly the form seems to be private (one-off) use: <a href="https://support.github.com/contact/feedback" rel="nofollow noopener">https://support.github.com/contact/feedback</a></p>

**Comment by kingthorin** [comment]
<p><a class="mention" href="/noinarisak">@noinarisak</a> I just came across this other syntax which is slightly better, but still requires a list of usernames.</p>
<pre><code class="lang-auto"> if: contains('["kingthorin","noinarisak"]', github.actor)
</code></pre>
<p>Ref: <a href="https://github.community/t/github-actions-branch-conditional/16057/7" class="inline-onebox">[GitHub Actions] branch conditional</a></p>
